source/PlayServicesResolver/src/ResolverVer1_1.cs

```diff
@@ -848,12 +858,26 @@ public override int Version()
                 DoResolutionUnsafe(svcSupport, destinationDirectory,
                                    unlockResolveAndSignalResolutionComplete);
             };
-            lock (resolveLock) {
-                resolveUpdateQueue.Enqueue(resolve);
-                EditorApplication.update += UpdateTryResolution;
+
+            if (runSynchronously)
+            {
+                resolve();
+            }else
+            {
+                lock (resolveLock)
+                {
+                    resolveUpdateQueue.Enqueue(resolve);
+                    EditorApplication.update += UpdateTryResolution;
+                }
             }
         }
 
+        public override void DoResolution(PlayServicesSupport svcSupport, string destinationDirectory)
+        {
+            runSynchronously = true;
```

Comment:
This seems a little dodgy to me as it's never set to false.  Once resolution has been run synchronously that's it all resolution will be synchronous.

Reply:
oh, i just assumed that one never reused an instance of the resolver. Will change it to pass the flag around as parameters instead then.